diesel/src/pg/metadata_lookup.rs

```diff
@@ -65,15 +72,27 @@ fn lookup_type(
             // skip the `$user` entry for now
             .filter(|f| !f.starts_with("\"$"))
             .map(|s| s.trim())
+            .filter(|&f| {
+                let is_temp = f == TEMP_SCHEMA_NAME;
+                search_path_has_temp_schema |= is_temp;
+                !is_temp
+            })
             .collect()
     };
 
-    let metadata = pg_type::table
+    let metadata_query = pg_type::table
         .inner_join(pg_namespace::table)
-        .filter(pg_namespace::nspname.eq(crate::dsl::any(search_schema)))
         .filter(pg_type::typname.eq(&cache_key.type_name))
-        .select((pg_type::oid, pg_type::typarray))
-        .first(conn)?;
+        .select((pg_type::oid, pg_type::typarray));
+    let nspname_filter = pg_namespace::nspname.eq(crate::dsl::any(search_schema));
+
+    let metadata = if search_path_has_temp_schema {
+        metadata_query
+            .filter(nspname_filter.or(pg_namespace::oid.eq(crate::dsl::sql("pg_my_temp_schema()"))))
```

Comment:
Could we define that function as ordinary sql function. `dsl::sql` will break the prepared statement caching here.